libraries/AP_HAL_ChibiOS/GPIO.cpp

```diff
@@ -202,3 +232,18 @@ void pal_interrupt_cb(void *arg)
     }
 }
 
+void pal_interrupt_cb_functor(void *arg)
+{
+    const uint32_t now = AP_HAL::micros();
+
+    struct gpio_entry *g = (gpio_entry *)arg;
+    if (g == nullptr) {
+        // what?
+        return;
+    }
+    if (g->fn == nullptr) {
```

Comment:
We should just do `if (!g->fn)`, comparing to nullptr makes it use a temporary (null) functor (compiler might be smart here, or not).